scrape_repo_information.py

```diff
@@ -0,0 +1,107 @@
+                lang_pair_name = type_content["name"]
+                #getting names
+                _, lg1, lg2 = lang_pair_name.split("-")
+                if lg1 != lg2:
+                #getting into repository
+                    link = "https://api.github.com/repos/apertium/%s/contents?" % lang_pair_name
+                    repo_json = json.loads(urllib.request.urlopen(link + params).read())
+                    for el in repo_json:
+                        if el["name"] == "modes.xml":
+                            download_url = el["download_url"]
+                            html_for_blob_utf_8 = str((urllib.request.urlopen(download_url)).read(), "utf-8")
+                            tree = xml.fromstring(html_for_blob_utf_8)
+                            for mode in tree:
+                                if lg1+"-"+lg2 == mode.attrib["name"] and "<" not in direction:
+                                    direction += "<"
+                                if lg2+"-"+lg1 in mode.attrib["name"] and ">" not in direction:
+                                    direction += ">"
+
+                        elif el["name"] == "apertium-%s.%s.dix" % ((lg1+"-"+lg2), (lg1+"-"+lg2)):
+                            stems = get_bidix_stem_count(el["download_url"])
+
+                        if direction == "><":
+                            direction = "<>"
+
+                    first_page_commits_link = "https://api.github.com/repos/apertium/%s/commits?" % lang_pair_name
+                    first_page_commits_resp = urllib.request.urlopen(first_page_commits_link + params)
+                    first_page_commits_data = first_page_commits_resp.read()
+                    first_page_commit_json = json.loads(first_page_commits_data)
+
+                    if first_page_commits_resp.info().get("Link") is not None:
+                        number_of_pages = first_page_commits_resp.info().get("Link").split(",")[1].split("page=")[1].split(">")[0]
+                    else:
+                        number_of_pages = 1
+
+                    last_page_commits_link = "https://api.github.com/repos/apertium/%s/commits?page=%s&" % (lang_pair_name, number_of_pages)
+                    last_page_commits_data = urllib.request.urlopen(last_page_commits_link + params).read()
+
+                    last_page_commit_json = json.loads(last_page_commits_data)
+                    last_updated = first_page_commit_json[0]["commit"]["committer"]["date"]
+                    created = last_page_commit_json[-1]["commit"]["committer"]["date"]
+
+                    pair = Pair(created=created, last_updated=last_updated, lg1=lg1, lg2=lg2, direction=direction, repo=repo_name, stems=stems)
+                    print(json.dumps(pair._asdict(), default=lambda o: dict(o)))
+                    pairs.append(pair)
+
+                    if CLIENT_SECRET == "" or CLIENT_ID == "":
+                        sleep(5)
+
+    if len(sys.argv) > 1:
+        with open(sys.argv[1], "w") as f:
```

Comment:
Instead of duplicating this code, just use something like `output_file = sys.argv[1] if len(sys.argv) > 1 else "pairs.json"`.